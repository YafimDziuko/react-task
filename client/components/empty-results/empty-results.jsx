import React from 'react';
import { connect } from 'react-redux';

import { setEmptyResults, returnToMainPage } from '../../action-creators';
import './empty-results.scss';

class EmptyResults extends React.Component {
    componentWillMount() {
        this.props.setEmptyResults();
    }

    render() {
        return (
            <p className="empty-page">No films found</p>
        );
    }
} 

const mapDispatchToProps = {
    setEmptyResults
}

const mapStateToProps = (state) => {
    return {
        user: state
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(EmptyResults);