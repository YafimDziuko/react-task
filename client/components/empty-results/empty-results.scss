.empty-page {
    margin: 0;
    padding-top: 30%;
    text-align: center;
    font-size: 32px;
    font-weight: 500;
}