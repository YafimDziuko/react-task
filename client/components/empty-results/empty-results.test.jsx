import React from 'react';
import { EmptyResults } from './empty-results';

describe('<EmptyResults/>', function() {
  it('should render Empty Results and match snapshot', function() {
      const wrapper = shallow(<EmptyResults />);
      expect(wrapper).toMatchSnapshot();
  });
});