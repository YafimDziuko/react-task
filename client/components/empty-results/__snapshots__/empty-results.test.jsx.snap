// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyResults/> should render Empty Results and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmptyResults />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "No films found",
      "className": "empty-page",
    },
    "ref": null,
    "rendered": "No films found",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "No films found",
        "className": "empty-page",
      },
      "ref": null,
      "rendered": "No films found",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
